via/views/debug.py

```diff
@@ -18,21 +18,23 @@ def debug_headers(_context, request):
     else:
         headers = clean_headers(request.headers)
 
+    self_url = request.route_url("debug_headers")
```

Comment:
Moved this up here to help debug why this was failing to begin with. `DummyRequest.route_url` was raising a pile of errors.

As a side effect, it also means we only call it the once... which is nice I guess?